### ISSUE TITLE
Renaming table column names doesn't persist in hte table visualization

### ISSUE THREAD

**Issue by TuringLovesDeathMetal** [lightdash/lightdash#3312]
### Description

If you rename a column in a table visualization, it doesn't persist (it gets replaced by the old name)

### To Reproduce

1. Create a table visualization
2. Update one of the column names
3. hit `save`
4. refresh the page
5. see the old column name pop up

**Comment by muhsinkamil** [comment]
Let me try this one @TuringLovesDeathMetal

**Comment by TuringLovesDeathMetal** [comment]
nice! go for it :)

**Comment by TuringLovesDeathMetal** [comment]
Another issue related to this: https://lightdash-community.slack.com/archives/C03MG2VTR08/p1664473252668129

User said that the naming is not persisting, but also whether or not a column is "hidden" is not persisting!

**Comment by hamzahc1** [comment]
Yes it does seem like the column set to "hidden" is also not persisting!

**Comment by muhsinkamil** [comment]
Bit of update: 
Seems data stored w.r to column config are stored a bit late ( Might be due to it is deeply nested, Still unsure of the cause ). This affects as described both column rename and hidden properties.

@ZeRego When I log this success data here: https://github.com/lightdash/lightdash/blob/main/packages/frontend/src/hooks/useSavedQuery.ts#L242, the column property in `chartConfig.config.columns` seems to be empty whereas in the network response it is available. 

Similarly, the [data in SavedExplorer](https://github.com/lightdash/lightdash/blob/main/packages/frontend/src/pages/SavedExplorer.tsx#L31) seems to miss `chartConfig.config.columns` property in the initial render, but appears to populate after few renders. Seems strange 🤔 

Have we had any similar issues with react-query before ?

**Comment by ZeRego** [comment]
Thanks for the initial investigation @muhsinkamil, this might be related to how VisualizationProvider manages state. This part of the code isn't great to be honest and not ideal for external contributors. 
I need to deep dive into the code to understand how to fix this.

**Comment by muhsinkamil** [comment]
Links to https://github.com/lightdash/lightdash/issues/3164, Hopefully this gets resolved in the refactor 🤞

**Comment by ZeRego** [comment]
@muhsinkamil you can look at my PR description to understand what was going wrong.
